src/abstract/CSBondLock.sol

```diff
@@ -59,6 +62,10 @@ abstract contract CSBondLock is ICSBondLock, Initializable {
         if (minBondLockRetentionPeriod > maxBondLockRetentionPeriod) {
             revert InvalidBondLockRetentionPeriod();
         }
+        // retention period can not be more than type(uint64).max to avoid overflow when setting bond lock
+        if (maxBondLockRetentionPeriod > type(uint64).max) {
```

Comment:
uint128?

Reply:
Within the BondLock structure, we have a field called retentionUntil. The value for this field is calculated as `currentTimestamp + retentionPeriod`, hence limiting to uint128 here will not be sufficient